File: nodejs/char_frequency.js

// count occurence of characters in string

var string = process.argv[2];

var charCounts = {};

string.split('').forEach(function(char) {
    if(charCounts[char]) {
        charCounts[char] = charCounts[char] + 1;
    } else {
        charCounts[char] = 1;
    }
});

for(var char in charCounts) {
    console.log(char + ':' + charCounts[char]);
}

File: python/char_frequency.py

# count occurence of characters in string

import sys

string = sys.argv[1]

charCounts = {}

for ch in string:
    if ch in charCounts:
        charCounts[ch] += 1
    else:
        charCounts[ch] = 1

for key, val in charCounts.iteritems():
    print key + ':' + str(val)
